prow/cmd/webhook-server/helpers.go

```diff
@@ -226,8 +234,8 @@ func createValidatingWebhookConfig(ctx context.Context, caPem string, client ctr
 				},
 				ClientConfig: admregistration.WebhookClientConfig{
 					Service: &admregistration.ServiceReference{
-						Namespace: "default",
-						Name:      "prowjob-validation-webhook",
+						Namespace: defaultNamespace,
+						Name:      prowjobAdmissionServiceName,
```

Comment:
These two values must match the service name and the namespace of where this webhook-service is deployed, for example in integration test they are https://github.com/kubernetes/test-infra/pull/27082/files#diff-8723ddc6f48050e3062d12a5c3587b5b39d05a8a8234cf19be071b6d3bf91ad0R21. Hardcoding the namespace and service names here means that for anyone who uses this `webhook-service`, it will need to be deployed in default namespace and named the same as const here. This is not very flexible, please make them configurable as command line args

Reply:
the service yaml is hardcoded though, so either ways the dnsName would have to match that. The namespace i could make a flag yes

Reply:
The service we are talking about, is the service that will be created for hosting `webhook-server`, it could be the same name as `prowjob-admission-webhook`, or could not. There is nothing stopping someone from creating a service that's named differently.

Reply:
in fact the rbac yaml is hardcoded as well

Reply:
yes that's fine, that's the contract the prow instance maintainer has to obey. It doesn't stop someone from naming it differently is what I want to say. There are many different prow instances managed by different companies, and each might have their own naming conventions on services, it's not user friendly to enforcing every single prow instance to name this service `prowjob-admission-webhook`